### ISSUE TITLE
Animation not working

### ISSUE THREAD

**Issue by ReindeerFloatilla** [cmbasnett/io_scene_abc#1]
We're looking to get animation support working with this plugin. So far it loads models from the game No One Lives Forever however it does not import/export animations very well. We are suspecting it doesn't have anything to do with the model format itself but rather our implementation in Blender.

We just need to figure out the matrix transformation steps and coordinate system translation. I can supply a model for testing purposes if necessary.

**Comment by technobaboo** [comment]
I'd love to work on this! I just need the model to test with it and I'll pull request if I can fix it!

**Comment by ReindeerFloatilla** [comment]
We've been using these models a lot for testing the plugin
[abc.zip](https://github.com/cmbasnett/io_scene_abc/files/2023635/abc.zip)

**Comment by ReindeerFloatilla** [comment]
We also have this model which is a cube, which we've used for testing purposes
[abc2.zip](https://github.com/cmbasnett/io_scene_abc/files/2023656/abc2.zip)

**Comment by cmbasnett** [comment]
Hi @jsking! Glad to hear you're going to take a crack at this. I've documented the model format [here](https://github.com/cmbasnett/io_scene_abc/wiki/ABC). Feel free to ask either of us if you run into a problem.

**Comment by technobaboo** [comment]
So I've taken a look at the animations and I'm guessing every single bone is disconnected from each other in the original game engine, as well as having a global rotation. Do you have any specifics on how Lithtech used those quaternions for bone rotations?

**Comment by cmbasnett** [comment]
Nothing in particular other than the somewhat out-of-sync documentation I was working off of, which is actually for an older version of the ABC format. (https://web.archive.org/web/20080605043638/http://bop-mod.com:80/download/docs/ABC-Format-v6.html).

I had a suspicion that the way I was constructing the bones (connected rather than disconnected) might have been the reason that the animations seemed to be completely bonkers when I applied the bone rotations. It might be instructive to be able to view the models inside of the ModelEdit program, since that cleared up a lot of questions I had. @ReindeerFloatilla will post the SDK files for you to have a look.

**Comment by ReindeerFloatilla** [comment]
[sdk.zip](https://github.com/cmbasnett/io_scene_abc/files/2024432/sdk.zip)

**Comment by ReindeerFloatilla** [comment]
we also ran modeledit through a debugger which helped us a lot with figuring out how it all worked

**Comment by technobaboo** [comment]
I've been working on this for several days and I still can't find how to debug windows programs. I'm using Linux, so I have to use WINE to run ModelEdit.

**Comment by cmbasnett** [comment]
I used IDA (Interactive Disassembler) as well as OllyDbg. However, these
really didn't help me all that much to be honest. Most of what I learned
was through mucking around with the model settings in Model Edit and
viewing the subsequent changes in the file being saved.

**Comment by technobaboo** [comment]
Ah okay then. How do you open the raw file? I tried text editors and they didn't like the file. Is it a special encoding?

**Comment by cmbasnett** [comment]
On Windows I use a program called HXD but there are lots of equivalent
programs for Unix systems (eg. http://www.wxhexeditor.org/).

**Comment by cmbasnett** [comment]
Also as a I mentioned before, the format of the file is described [here](https://github.com/cmbasnett/io_scene_abc/wiki/ABC).

**Comment by SmiteIsTrashBro** [comment]
I'm working in something similar but for the noesis instead, to export Aliens vs. Predator 2 models. Your informations was actually useful, thank you. For now I've nothing more than model meshes and its bones, I'm stuck in figure out how export the animations as you, so to say. If I do it, I'll share the solution with you, but until then... do you have any news?

Besides, this .lta (converted .abc model) may help you, since it can be open in simple text editor.
https://mega.nz/#!iYtRlQYR!00DhYMZ0QValf95v1MTMI3bsmEwlQ7xB4gbXspJSVOs

**Comment by ReindeerFloatilla** [comment]
Unfortunately no there isn't any updates. Making any sort of .abc plugin regardless of Blender or Noesis would be extremely great. 

AVP2 used the same version of .abc as No One Lives Forever. And we discovered that the format was just added onto the Lithtech 1.0 version of the format. So it would theoretically be possible to also get it to work with Blood 2 and Shogo.

We mostly talk on Discord. If you wanna chat there it'd be great.

**Comment by SmiteIsTrashBro** [comment]
Yes, would be. My main purpose it's AVP but I'm almost sure there are people interested in the other games you cited.

Yes please, can you inform invite link? Or add me there,  CrazY.#5317

**Comment by haekb** [comment]
WIP PR is here https://github.com/cmbasnett/io_scene_abc/pull/2 

If anyone has any ideas on the bone twisting rotations, I'd be happy to hear!

**Comment by haekb** [comment]
Oh just realized my WIP PR was merged. 

Just a heads up, while it does import animations (not well), I haven't tested exporting on blender 2.80 🤔

I've played around with different ways of applying the animations and I haven't had any luck on that front.
